### ISSUE TITLE
Better/More efficient Keyboard accessibility for TimePicker dialog.

### ISSUE THREAD

**Issue by Jonas-Sander** [flutter/flutter#71999]
![Flutter timepicker accessability](https://user-images.githubusercontent.com/29028262/101649531-4892e500-3a3b-11eb-816e-cc98000420a5.gif)

## Use case
We have a web app where users can add events to an calender.  
The user needs to pick a start and end time for this event.  
We want this time picking to be accessible by keyboard and also be quick an efficient. 

My current problems with the design:
* The first (hour) field is not automatically focused when the dialog opens.  This means the user will have to always press TAB one time before he can enter/edit the input.
* With `showTimePicker` `initialTime` is required.\
This means the user will **always** have to first delete the initial input before he inserts his own time.
* To confirm the input the user will have to press "TAB" multiple times to get to the "OK" button and the press space to confirm.

## Proposal
* The `initalTime` should not be required, instead there may be a "hint time" with low opacity.\
or
* When focusing the hour and minutes field the existing input should be selected by default (see picture). In this case the useer can just overwrite the existing input by just typing in their own. 
<img width="319" alt="Bildschirmfoto 2020-12-09 um 16 19 42" src="https://user-images.githubusercontent.com/29028262/101648784-77f52200-3a3a-11eb-819b-26909e0c8d25.png">

* When `showTimePicker` is called the first field (hour) should be focused automatically instead of having the user press TAB or click into the field.
* When pressing enter in the hour field the user focus should change to the minutes field (same behavior as pressing space in the hour field)
* When pressing "Enter" in the minutes field the dialog should successfully close (same as pressing the "OK" button). With this change users wont have to repeatedly press TAB to get the focus to the "OK" button. 

For the last two bullet points see also: #71998

**Comment by HansMuller** [comment]
CC @darrenaustin

**Comment by JanezStupar** [comment]
Anything going on on this? I went through the pub packages today and there is really no nice time picker option at the moment.

It would be nice if we either got a better time picker OR if we could inject our own.

Overall look and feel of current picker is okay.

Would love if I could modify it so that when entering a input field the value would get selected so it could get instantly replaced.

I would like it if the input field would not allow asinine values such as "39" in hour field.

Another thing I would like is if I could get say up and down button on keyboard to map to "increase" and "decrease" values.
